pgmax/fg/groups.py

```diff
@@ -761,14 +815,35 @@ def _get_variables_to_factors(
                             self.variable_group[self.variable_names_for_factors[ii]]
                         ),
                         configs=self.factor_configs,
-                        log_potentials=log_potentials[ii],
+                        log_potentials=self.factor_group_log_potentials_full[ii],
                     ),
                 )
                 for ii in range(len(self.variable_names_for_factors))
             ]
         )
         return variables_to_factors
 
+    def compile_wiring(
+        self, vars_to_starts: Mapping[nodes.Variable, int]
+    ) -> enumeration.EnumerationWiring:
+        """Compile EnumerationWiring for the EnumerationFactorGroup
+
+        Args:
+            vars_to_starts: A dictionary that maps variables to their global starting indices
+                For an n-state variable, a global start index of m means the global indices
+                of its n variable states are m, m + 1, ..., m + n - 1
+
+        Returns:
+             EnumerationWiring for the EnumerationFactorGroup
+        """
+        return _compile_factor_group_wiring(
```

Comment:
Can we keep all the relevant logic inside this function, instead of defining a function elsewhere?